src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/FilingService.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.companieshouse.confirmationstatementapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.confirmationstatementapi.exception.SubmissionNotFoundException;
+import uk.gov.companieshouse.confirmationstatementapi.model.json.ConfirmationStatementSubmissionJson;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class FilingService {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+    private ConfirmationStatementService confirmationStatementService;
+    private Environment environment;
+
+    @Autowired
+    public FilingService(ConfirmationStatementService csService, Environment env) {
+        confirmationStatementService = csService;
+        environment = env;
+    }
+
+    public FilingApi generateConfirmationFiling(String confirmationStatementId) throws SubmissionNotFoundException {
+        FilingApi filing = new FilingApi();
+        filing.setKind("Confirmation Statement");
+        setFilingApiData(filing, confirmationStatementId);
+        setDescription(filing);
+        return filing;
+    }
+
+    private void setFilingApiData(FilingApi filing, String confirmationStatementId) throws SubmissionNotFoundException {
+        Optional<ConfirmationStatementSubmissionJson> submissionOpt =
+                confirmationStatementService.getConfirmationStatement(confirmationStatementId);
+        if(submissionOpt.isEmpty()){
+            throw new SubmissionNotFoundException(
+                    String.format("Empty submission returned when generating filing for %s", confirmationStatementId));
+        }
+        ConfirmationStatementSubmissionJson submission = submissionOpt.get();
```

Comment:
line 44 - sonarLint plugin has highlighted that we need to call submissionOpt.isPresent() before accessing the value.

Reply:
We are doing in the method we call. I didn' t think I'd need to do it again.

Reply:
we could use the orElseThrow option - it seems to clear up the sonarLint issue

ConfirmationStatementSubmissionJson submission = submissionOpt
                .orElseThrow(() ->
                        new SubmissionNotFoundException(
                                String.format("Empty submission returned when generating filing for %s", confirmationStatementId)));